lib/semantics/resolve-names.cc

```diff
@@ -1228,16 +1232,15 @@ void ScopeHandler::EraseSymbol(const SourceName &name) {
   currScope().erase(name);
 }
 
-void ScopeHandler::ApplyImplicitRules(const SourceName &name, Symbol &symbol) {
-  if (symbol.has<UnknownDetails>()) {
-    symbol.set_details(ObjectEntityDetails{});
-  }
-  if (auto *details{symbol.detailsIf<ObjectEntityDetails>()}) {
-    if (!details->type()) {
-      if (const auto type{GetImplicitType(symbol)}) {
-        details->set_type(*type);
-      }
-    }
+void ScopeHandler::ApplyImplicitRules(Symbol &symbol) {
+  ConvertToObjectEntity(symbol);
```

Comment:
Does GetType return a boolean?  Are we ok with non-boolean functions being used as booleans?

Reply:
It's a pointer. I'm okay with requiring `if (symbol.GetType() != nullptr)` if that's the style we want to follow.

Reply:
If a function result value is being used only to determine a predicate, implicit conversion to `bool` is fine by me.